chain/epoch_manager/src/lib.rs

```diff
@@ -445,6 +452,12 @@ impl EpochManager {
         // This epoch info is computed for the epoch after next (T+2),
         // where epoch_id of it is the hash of last block in this epoch (T).
         self.save_epoch_info(store_update, &next_next_epoch_id, next_next_epoch_info)?;
+
+        // Store next_epoch info again since sandbox changes it:
+        if cfg!(feature = "sandbox") {
+            self.save_epoch_info(store_update, &next_epoch_id, next_epoch_info)?;
+        }
```

Comment:
This logic looks quite dubious to me. Why do you need to change epoch info retroactively? Also we should have unit test for this.

Reply:
So, there's a clone that happens when we grab `next_epoch_info`. This is why we need to update it again. I'll try to add a unit test for this